test/unit/constraints/array.test.ts

```diff
@@ -0,0 +1,113 @@
+import "../metadataShim";
+import {registerJoi, WORKING_SCHEMA_KEY} from "../../../src/core";
+import * as Joi from 'joi';
+import { Max, Length, ArraySchema, Min, Unique } from "../../../src/constraints/array";
+import {Validator} from "../../../src/Validator";
+import { testConstraint } from "../testUtil";
+
+
+registerJoi(Joi);
+
+describe('Array Test', () => {
+    describe('Test Array methods ', () => {
```

Comment:
⛏ Try to make the test descriptions as accurate as possible. "Method" normally describes a function bound to a class instance. We don't really need to say that we're testing Array constraint decorators, because that's what the outer `describe` description will tell us.

I'd suggest moving the `it()` test cases out of this `describe` block, and adding one `describe` block per decorator under test. So, `@ArraySchema()` is in its own describe block, `@Max()` is in its own describe block, etc.